src/plugins/thirdroom/thirdroom.game.ts

```diff
@@ -488,14 +489,15 @@ async function onGLTFViewerLoadGLTF(ctx: GameState, message: GLTFViewerLoadGLTFM
 }
 
 function disposeWorld(worldResource: RemoteWorld) {
+  console.log("===== disposeWorld ======");
   worldResource.activeCameraNode = undefined;
   worldResource.activeAvatarNode = undefined;
   worldResource.environment = undefined;
   worldResource.firstNode = undefined;
 }
 
 async function loadEnvironment(ctx: GameState, url: string, scriptUrl?: string, fileMap?: Map<string, string>) {
-  disposeWorld(ctx.worldResource);
+  // disposeWorld(ctx.worldResource);
```

Comment:
Unfortunately I think this also breaks when you swap the scene via uploading a new scene as you don't exit the scene when it happens. I think the solution will require some fixes to the react / world loading lifecycle.

Reply:
hmm ok good point. will investigate that case